format/src/matbin/mod.rs

```diff
@@ -0,0 +1,309 @@
+use std::{borrow::Cow, io};
+
+use bytemuck::PodCastError;
+use byteorder::LE;
+use thiserror::Error;
+use widestring::{U16Str, U16String};
+use zerocopy::{FromBytes, FromZeroes, Ref, F32, U32, U64};
+
+use crate::io_ext::zerocopy::Padding;
+
+#[derive(Debug, Error)]
+pub enum MatbinError {
+    #[error("Could not copy bytes {0}")]
+    IO(#[from] io::Error),
+
+    #[error("Could not read string")]
+    String(#[from] ReadUtf16StringError),
+
+    #[error("Got unknown parameter type {0}")]
+    UnknownParameterType(u32),
+
+    #[error("Could not create reference to value")]
+    ValueZerocopy,
+}
+
+#[derive(FromZeroes, FromBytes)]
+#[repr(C)]
+#[allow(unused)]
+// Defines a material for instancing in FLVERs and such.
+// It does so by pointing at a shader and specifying the parameter/sampler
+// setup.
+pub struct Matbin<'buffer> {
```

Comment:
I don't think we need to name lifetimes when there's only one present and its bounds are obvious (e.g. everything is tied to the lifetime of the buffer).

Reply:
Renamed per 808b7a96a9af4984a4ac229712b30be1cc015e48